### PR TITLE
App Store subscriptions specification

### PR DESCRIPTION
**Pull request by SerKnight**
Implements the endpoints available in Xero's API spec.

## Description

https://developer.xero.com/documentation/guides/how-to-guides/xero-app-store-subscriptions/

https://developer.xero.com/documentation/api/xero-app-store/subscriptions

## Release Notes
Subscriptions endpoints enable you to query data about the users who have paid to subscribe to your application through Xero.

* adds `EASTERISLANDSTANDARDTIME`

## Types of Changes
- [x] New feature (non-breaking change that adds functionality)

### CODE REVIEW

File: xero-app-store.yml [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+          items:
+            $ref: "#/components/schemas/Plan"
+        startDate:
+          description: Date when the subscription was first created.
+          format: date-time
+          type: string
+        status:
+          description: Status of the subscription. Available statuses are ACTIVE, CANCELED, and PAST_DUE.
+          type: string
+        testMode:
```
Comment (rdemarco-xero) [inline xero-app-store.yml:134]: Based on the [docs ](https://developer.xero.com/documentation/api/xero-app-store/subscriptions#get-subscription), 
 this param's name should be "isTestMode".

File: xero-app-store.yml [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+      type: object
+      properties:
+        currentPeriodEnd:
+          description: |
+            End of the current period that the subscription has been invoiced for.
+          format: date-time
+          type: string
+        endDate:
+          description: If the subscription has been canceled, this is the date when the subscription ends. If null, the subscription is active and has not been cancelled
+          format: date-time
```
Comment (rdemarco-xero) [inline xero-app-store.yml:112]: I haven't tested this API yet. Do we know for sure that we return a simple date-time format and not the MS date format used in our other APIs (ex. "/Date(1476316800000+0000)/")?

**Review by rdemarco-xero — DISMISSED** [review]
I believe "testMode" needs to be renamed to "isTestMode".

**Comment by SerKnight** [comment]
> I believe "testMode" needs to be renamed to "isTestMode".

I thought that too as that is what the html docs say, but the API response is just `testMode` @sethreidnz can you confirm?

File: xero-app-store.yml (hunk 2 above)
Comment (SerKnight) [reply xero-app-store.yml:112]: It does look that way. In newer API's I have not seen the ms date pattern, but would be good to know from the core eng team.

@sethreidnz 

Screen Shot 2021-08-04 at 11 21 39 AM

File: xero-app-store.yml [hunk 3]
```diff
@@ -0,0 +1,260 @@
+openapi: 3.0.0
+info:
+  version: "2.14.0"
```
Comment (RettBehrens) [inline xero-app-store.yml:3]: this version is behind the latest release

File: xero-app-store.yml [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Subscription"
+              example: '
+                        {
+                          "currentPeriodEnd": "2021-07-20T03:13:48Z",
+                          "endDate": "2021-07-20T03:13:48Z",
+                          "id": "01b5a6f4-8936-4bfa-b703-830702312b87",
+                          "isTestMode": true,
```
Comment (RettBehrens) [inline xero-app-store.yml:38]: we should confirm if it's `isTestMode` or `testMode` and update the docs example to match

File: xero-app-store.yml [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+      security:
+        - OAuth2: [marketplace.billing]
+components:
+  securitySchemes:
+    OAuth2:
+      type: oauth2
+      description: For more information
+      flows:
+        authorizationCode:
+          authorizationUrl: "https://login.xero.com/identity/connect/authorize"
```
Comment (RettBehrens) [inline xero-app-store.yml:83]: I thought app store billing used client credentials grant - do we need the auth URL here?

File: xero-app-store.yml (hunk 3 above)
Comment (RettBehrens) [reply xero-app-store.yml:3]: current release is 2.14.2 per github

File: xero_files.yaml [hunk 6]
```diff
@@ -4,7 +4,7 @@ servers:
     url: https://api.xero.com/files.xro/1.0/
 info:
   title: Xero Files API
-  version: "2.13.5"
+  version: "2.14.0"
```
Comment (RettBehrens) [inline xero_files.yaml:7]: current release is 2.14.2 per github

File: xero_bankfeeds.yaml [hunk 7]
```diff
@@ -1,6 +1,6 @@
 openapi: 3.0.0
 info:
-  version: "2.13.5"
+  version: "2.14.0"
```
Comment (RettBehrens) [inline xero_bankfeeds.yaml:3]: current release is 2.14.2 per github

File: xero_assets.yaml [hunk 8]
```diff
@@ -1,6 +1,6 @@
 openapi: 3.0.0
 info:
-  version: "2.13.5"
+  version: "2.14.0"
```
Comment (RettBehrens) [inline xero_assets.yaml:3]: current release is 2.14.2 per github

File: xero_accounting.yaml [hunk 9]
```diff
@@ -1,7 +1,7 @@
 openapi: 3.0.0
 info:
   title: Xero Accounting API
-  version: "2.13.5"
+  version: "2.14.0"
```
Comment (RettBehrens) [inline xero_accounting.yaml:4]: current release is 2.14.2 per github

File: xero-projects.yaml [hunk 10]
```diff
@@ -1,6 +1,6 @@
 openapi: 3.0.0
 info:
-  version: "2.13.5"
+  version: "2.14.0"
```
Comment (RettBehrens) [inline xero-projects.yaml:3]: current release is 2.14.2 per github

File: xero-payroll-uk.yaml [hunk 11]
```diff
@@ -1,6 +1,6 @@
 openapi: 3.0.0
 info:
-  version: '2.13.5'
+  version: '2.14.0'
```
Comment (RettBehrens) [inline xero-payroll-uk.yaml:3]: current release is 2.14.2 per github

File: xero-payroll-nz.yaml [hunk 12]
```diff
@@ -1,6 +1,6 @@
 openapi: 3.0.0
 info:
-  version: '2.13.5'
+  version: '2.14.0'
```
Comment (RettBehrens) [inline xero-payroll-nz.yaml:3]: current release is 2.14.2 per github

File: xero-payroll-au.yaml [hunk 13]
```diff
@@ -1,6 +1,6 @@
 openapi: 3.0.0
 info:
-  version: '2.13.5'
+  version: '2.14.0'
```
Comment (RettBehrens) [inline xero-payroll-au.yaml:3]: current release is 2.14.2 per github

File: xero-identity.yaml [hunk 14]
```diff
@@ -1,6 +1,6 @@
 openapi: 3.0.0
 info:
-  version: "2.13.5"
+  version: "2.14.0"
```
Comment (RettBehrens) [inline xero-identity.yaml:3]: current release is 2.14.2 per github

File: xero-app-store.yml (hunk 5 above)
Comment (SerKnight) [reply xero-app-store.yml]: yes you are correct - I updated this to match actuality. This isn't technically used in the SDK Codegen but good to represent it accurately.

File: xero-app-store.yml (hunk 4 above)
Comment (SerKnight) [reply xero-app-store.yml:38]: Screen Shot 2021-08-05 at 1 49 51 PM

Json coming back from API is `testMode`

File: xero-app-store.yml (hunk 1 above)
Comment (SerKnight) [reply xero-app-store.yml:133]: I don't think docs are accurate in this case. Can always update if the API actually changes to reflect that though.

File: xero-app-store.yml (hunk 4 above)
Comment (RettBehrens) [reply xero-app-store.yml:38]: sounds like we confirmed this should be `testMode` in the example block instead of `isTestMode`

**Review by RettBehrens — COMMENTED** [review]
looks good to go pending the change of `isTestMode` to `testMode`

**Review by RettBehrens — APPROVED** [review]

**Pull request merged**